R/default_cdisc_join_keys.R

```diff
@@ -34,3 +34,35 @@ build_cdisc_join_keys <- function(default_cdisc_keys) {
   }
   jk
 }
+
+
+#' Function that returns the default keys for a `CDISC` dataset by name
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' @param dataname name of the `CDISC` dataset
+#'
+#' @return \code{keys} object
+#'
+#' @export
+#'
+#' @examples
+#' get_cdisc_keys("ADSL")
+get_cdisc_keys <- function(dataname) {
+  checkmate::assert_string(dataname)
+
+  if (!(dataname %in% names(default_cdisc_keys))) {
+    stop(paste(sprintf("get_cdisc_keys does not support datasets called %s", dataname),
+      "  Please specify the keys directly, for example:",
+      sprintf(
+        "  cdisc_dataset(dataname = \"%s\", keys = c(\"STUDYID\", \"USUBJID\", ...), parent = \"ADSL\", ...)",
+        dataname
+      ),
+      sep = "\n"
+    ))
+  } else {
+    cdisc_keys <- default_cdisc_keys[[dataname]]$primary
+
+    return(cdisc_keys)
+  }
+}
```

Comment:
Function is also defined in `data_label.R`. One of them should be removed.

Should it be called `get_cdisc_primary_keys` ?

Reply:
I'm removing this at all and use `default_cdisc_join_keys["ADSL", "ADSL"]`. It is used just in few files in the whole organization

Reply:
Great!! One occurence left on https://github.com/insightsengineering/user-guide/blob/477b72fbae1b172fbde30e89d5c0704ed80301f7/content/teal/using_study_data.md?plain=1#L59

What's the process for that repo, can we just change it now or should we wait till the release?

Reply:
Whaaat, what is it! I'll probably remove it. It's redundant I guess

Reply:
This repo has been update last time 2 years ago. Probably has failures already. It is not a blocker.